site/_posts/2017-11-27-ray-0.3-release.markdown

```diff
@@ -0,0 +1,158 @@
+---
+layout: post
+title: "Ray: 0.3 Release"
+excerpt: "This post announces the release of Ray 0.3."
+date: 2017-11-29 00:00:00
+---
+
+We are pleased to announce the Ray 0.3 release. This release includes a number
+of bug fixes, as well as [**distributed actors**][1] and a [**hyperparameter
+search library**][2] (including integration with [**RLlib**][3]).
+
+To upgrade to the latest version, run
+
+```
+pip install -U ray
+```
+
+## Hyperparameter Search Tool
+
+This release adds [**Ray.tune**][2], a distributed hyperparameter evaluation
+tool for long-running tasks such as reinforcement learning and deep learning
+training. It currently includes the following features:
+
+- Pluggable **early stopping algorithms** including the Median Stopping Rule and
+  [Hyperband][5].
+- Integration with **visualization tools** such as [TensorBoard][6],
+  [rllab's VisKit][7], and a [parallel coordinates visualization][8].
+- Flexible **trial variant generation**, including grid search, random search,
+  and conditional parameter distributions.
+- **Resource-aware scheduling**, including support for concurrent runs of
+  algorithms that require GPUs or are themselves parallel and distributed.
+
+Ray.tune provides a Python API for use with deep learning and other compute
+intensive training tasks. Here is a toy example illustrating usage.
+
+```python
+from ray.tune import register_trainable, grid_search, run_experiments
+
+def my_func(config, reporter):
+    import time, numpy as np
+    i = 0
+    while True:
+        reporter(timesteps_total=i, mean_accuracy=i ** config['alpha'])
+
+        i += config['beta']
+        time.sleep(0.01)
+
+register_trainable('my_func', my_func)
+
+run_experiments({
+    'my_experiment': {
+        'run': 'my_func',
+        'resources': {'cpu': 1, 'gpu': 0},
+        'stop': {'mean_accuracy': 100},
+        'config': {
+            'alpha': grid_search([0.2, 0.4, 0.6]),
+            'beta': grid_search([1, 2]),
+        },
+    }
+})
+```
+
+In-progress results can be visualized live using tools such as Tensorboard and
+rllab's VisKit (or you can read the JSON format logs directly from the driver
+node):
+
+View the [documentation][2] and the [code][9].
+
+### RLlib integration with Ray.tune
```

Comment:
would it make sense to swap the order, i.e., **Ray.tune integration with RLlib**